src/common/inference/mod.rs

```diff
@@ -6,3 +12,41 @@ pub mod query_requests_grpc;
 pub mod query_requests_rest;
 pub mod service;
 pub mod update_requests;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct InferenceToken(pub Option<String>);
+
+impl InferenceToken {
+    pub fn new(key: String) -> Self {
```

Comment:
Is there nothing to validate about the token String itself?

Reply:
Do we want to allow an empty string?